keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -289,7 +290,7 @@ class SecurityGroupHandler(
                   )
                   else -> ReferenceRule(
                     protocol,
-                    if (ingressGroup.name == name) null else ingressGroup.name, // if it's a self-referential rule keel models the name as null
+                    ingressGroup.name,
```

Comment:
No longer necessary to null out the value coming back from River if it's the same as the current security group name